sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -287,23 +291,36 @@ def to_arrow(self) -> pyarrow.Table:
 
 
 def block_until_done(client, bq_job):
-    def _is_done(job_id):
-        return client.get_job(job_id).state in ["PENDING", "RUNNING"]
+    """
+    Waits a maximum of 30 minutes for bq_job to finish running.
+    Args:
+        client: A bigquery.client.Client to monitor the bq_job.
```

Comment:
Thanks for the doc. Very useful.
[Optional] we could also add python types to the input of this function. As you want

Reply:
I prefer adding type hints as well.